### PR TITLE
Run3-Sim154 Try to avoid memory leak issues in SimG4Core/PrintGeomeInfo and Geometry/TrackerCommonData

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Try to avoid memory leak issues in SimG4Core/PrintGeomeInfo and Geometry/TrackerCommonData

#### PR validation:

Use the runTheMatrix test workflows 

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: SimG4Core/PrintGeomInfo/plugins/PrintMaterialBudgetInfo.cc [hunk 1]
```diff
@@ -66,7 +67,7 @@ class PrintMaterialBudgetInfo : public SimWatcher,
   std::string name;
   int nchar;
   mpvpv thePVTree;
-  G4VPhysicalVolume* theTopPV;
+  //  std::unique_ptr<G4VPhysicalVolume> theTopPV;
```
Comment (civanch) [inline SimG4Core/PrintGeomInfo/plugins/PrintMaterialBudgetInfo.cc:70]: @bsunanda , we use not to have commented code.

File: SimG4Core/PrintGeomInfo/plugins/PrintMaterialBudgetInfo.cc [hunk 2]
```diff
@@ -229,7 +231,7 @@ void PrintMaterialBudgetInfo::dumpHierarchyLeaf(
   std::set<G4LogicalVolume*> lvDaughters;
   int NoDaughters = lv->GetNoDaughters();
   while ((NoDaughters--) > 0) {
-    G4VPhysicalVolume* pvD = lv->GetDaughter(NoDaughters);
+    G4VPhysicalVolume* pvD = (lv->GetDaughter(NoDaughters));
```
Comment (civanch) [inline SimG4Core/PrintGeomInfo/plugins/PrintMaterialBudgetInfo.cc:234]: @bsunanda , nothing changed

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by civanch** [comment]
+1

**Comment by bsunanda** [comment]
@rappoccio, @antoniovilela Please integrate this

**Comment by antoniovilela** [comment]
+1

**Pull request merged**